wa/workloads/antutu/__init__.py

```diff
@@ -109,16 +146,109 @@ def extract_scores(self, context, regex_version):
                         entry = regex.pattern.rsplit(None, 1)[0]
                         context.add_metric(entry, result, lower_is_better=False)
+                        #Calculate group scores if 'CPU' in entry:
+                        if 'CPU' in entry:
+                            cpu.append(result)
+                            cpu_result = sum(cpu)
+                        if 'GPU' in entry:
+                            gpu.append(result)
+                            gpu_result = sum(gpu)
+                        if any([i in entry for i in ['Data', 'Document', 'Image', 'User', 'Video']]):
+                            ux.append(result)
+                            ux_result = sum(ux)
+                        if any([i in entry for i in ['RAM', 'ROM']]):
+                            mem.append(result)
+                            mem_result = sum(mem)
                         expected_results -= 1
         if expected_results > 0:
             msg = "The Antutu workload has failed. Expected {} scores, Detected {} scores."
             raise WorkloadError(msg.format(len(regex_version), expected_results))
 
+        context.add_metric('CPU Total Score', cpu_result, lower_is_better=False)
+        context.add_metric('GPU Total Score', gpu_result, lower_is_better=False)
+        context.add_metric('UX Total Score', ux_result, lower_is_better=False)
+        context.add_metric('MEM Total Score', mem_result, lower_is_better=False)
+
+        #Calculate overall scores
+        overall_result = float(cpu_result + gpu_result + ux_result + mem_result)
+        context.add_metric('Overall Score', overall_result, lower_is_better=False)
+
     def update_output(self, context):
         super(Antutu, self).update_output(context)
+        if self.version.startswith('10'):
+            self.extract_scores(context, self.regex_matches_v10)
         if self.version.startswith('9'):
             self.extract_scores(context, self.regex_matches_v9)
         if self.version.startswith('8'):
             self.extract_scores(context, self.regex_matches_v8)
         if self.version.startswith('7'):
             self.extract_scores(context, self.regex_matches_v7)
+
+
+class AntutuBDP(Workload):
+
+    name = "antutu_bdp"
+    description = '''
+    Workload for executing the BDP versions of the Antutu APK.
+
+    This will only work with specific APKS provided by Antutu but does
+    unlock command line automation and the capturing of a result file
+    as opposed to using UiAuto and Regex.
+
+    Known working version: 10.4.3-domesticAndroidFullBdp
+    '''
+    package_name = 'com.antutu.ABenchMark'
+    activity = 'com.android.module.app.ui.start.ABenchMarkStart --ez isExternal true --es whereTo "test"'
+
+    def initialize(self, context):
+        super(AntutuBDP, self).initialize(context)
+        #Set the files and directories we need
+        self.test_dir = os.path.join(self.target.external_storage_app_dir, 'com.antutu.ABenchMark/files/.antutu/')
```

Comment:
Could this be updated to use the path join functionality from the target for the full path? 
i.e. `self.taraget.path.join(self.target.external_storage_app_dir, 'com.antutu.ABenchMark', 'files', '.antutu')`

And the same for the below paths.